### ISSUE TITLE
Allow ExEx provider to subscribe to safe & finalized block updates

### ISSUE THREAD

**Issue by unitezen** [paradigmxyz/reth#10316]
### Describe the feature

The FullProvider you get from ExExContext's components doesn't implement ForkChoiceSubscriptions.

We need to modify a few things, which I have made a separate PR for each feature:
1. #10317
2. #10318
3. #10319

I'm not sure how to properly pull request separate features like these, where 1 must be merged before 2 and 2 before 3. I've branched them out each with 1, 1&2, 1&2&3. Please kindly suggest any fix if necessary.
